firebase-firestore/src/test/java/com/google/firebase/firestore/remote/RemoteSerializerTest.java

```diff
@@ -611,7 +616,80 @@ public void testEncodesMultipleFiltersOnDeeperCollections() {
                                             .setField(
                                                 FieldReference.newBuilder().setFieldPath("tags"))
                                             .setOp(Operator.ARRAY_CONTAINS)
-                                            .setValue(valueBuilder().setStringValue("pending"))))))
+                                            .setValue(valueBuilder().setStringValue("pending"))))
+                            .addFilters(
+                                Filter.newBuilder()
+                                    .setFieldFilter(
+                                        FieldFilter.newBuilder()
+                                            .setField(
+                                                FieldReference.newBuilder().setFieldPath("tags"))
+                                            .setOp(Operator.IN)
+                                            .setValue(valueBuilder().setArrayValue(array))))))
+            .addOrderBy(
+                Order.newBuilder()
+                    .setField(FieldReference.newBuilder().setFieldPath("prop"))
+                    .setDirection(Direction.ASCENDING))
+            .addOrderBy(defaultKeyOrder());
+    QueryTarget.Builder queryBuilder =
+        QueryTarget.newBuilder()
+            .setParent("projects/p/databases/d/documents/rooms/1/messages/10")
+            .setStructuredQuery(structuredQueryBuilder);
+    Target expected =
+        Target.newBuilder()
+            .setQuery(queryBuilder)
+            .setTargetId(1)
+            .setResumeToken(ByteString.EMPTY)
+            .build();
+
+    assertEquals(expected, actual);
+    assertEquals(serializer.decodeQueryTarget(serializer.encodeQueryTarget(q)), q);
+  }
+
+  @Test
+  public void testEncodesDisjunctiveFiltersOnDeeperCollections() {
+    Query q =
+        Query.atPath(ResourcePath.fromString("rooms/1/messages/10/attachments"))
+            .filter(filter("prop", "<", 42))
+            .filter(filter("author", "==", "dimond"))
+            .filter(filter("tags", "array-contains-any", asList("pending", "dimond")));
```

Comment:
So...  
1) This test claims to test "disjunctive filters" but I think only tests array-contains-any, not in, right?
2) This test has a ton of unnecessary duplication of the previous test (we don't need to test <, ==, or order-bys again).

I think what happened is Android for whatever reason originally had one ginormous test instead of individual tests.  But now that we're trying to test filters that can't be combined, that's biting us.  So I would try to create more isolated minimal tests, ideally moving us closer to being in-sync with the way the tests work on TypeScript (and probably iOS though I haven't looked).